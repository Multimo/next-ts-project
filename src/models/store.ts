import { types, flow } from 'mobx-state-tree';

// isomophic fetch since fetch only works only exists on the window
import fetch from 'node-fetch';


export const getAlbums = async () => {
  try {
    const res = await fetch('https://jsonplaceholder.typicode.com/albums');
    const albums = await res.json();
    return albums;
  } catch (e) {
    console.log(e);
  }
}

export const getPosts = async () => {
  try {
    const res = await fetch('https://jsonplaceholder.typicode.com/posts');
    const posts = await res.json();
    return posts;
  } catch (e) {
    console.log(e);
  }
}

const Album = types.model('Album' ,{
  id: types.number,
  userId: types.number,
  title: types.string,
})

const Post = types.model('Post' ,{
  id: types.number,
  userId: types.number,
  title: types.string,
})

export const albumStore = types
.model('store', {
  albums: types.optional(types.array(Album), []),
  posts: types.optional(types.array(Post), []),
  toggle: types.optional(types.boolean, false),
})
.actions((self) => ({
  toggler() {
    self.toggle = !self.toggle;
    },
    afterCreate: async () => {
      await getAlbums();
      await getPosts();
    },
  }));
  // const fetchAlbums = flow(function* () {
    //   const res = yield getAlbums();
    //   console.log('FETCHED!!!!!!!!!!!!!!!!!')
    //   applySnapshot(self.albums, res)
    // });
    
    // return ({
  
        //   fetchAlbums,
        
        // });
        // });
        
        
        // export const rehydrateStore = async (isServer: boolean, snapshot = null) => {
          //   if (isServer || alStore.albums.length === 0) {
            //     const hellomyoldfirend = await getAlbums();
            //     const posts = await getPosts();
            //     console.log()
            //     store = albumStore.create({
              //       albums: hellomyoldfirend || [],
              //       posts: posts || [],
              //     });
              //     return { 
                //       initialState: getSnapshot(store), 
                //       isServer, 
                //     };
                //   }
                
                
                //   // if (snapshot) {
                  //   //   console.log('is store alive?', alStore, snapshot)
                  //   //   return store
                  //   // }
                  //     // store = applySnapshot(alStore, snapshot)
                  
                  //   return alStore;
                  // };
export const alStore = albumStore.create();
                  

