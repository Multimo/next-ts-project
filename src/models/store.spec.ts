import { albumStore } from './store';


describe('Store unit tests', () => {
  it('should match snapshot when created', () => {
    albumStore.create();
    return expect(albumStore).toMatchSnapshot();
  })
})