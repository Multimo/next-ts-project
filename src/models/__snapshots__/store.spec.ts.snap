// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store unit tests should match snapshot when created 1`] = `
ModelType$$1 {
  "flags": 128,
  "identifierAttribute": undefined,
  "initializers": Array [
    [Function],
    [Function],
  ],
  "isType": true,
  "name": "store",
  "postProcessor": undefined,
  "preProcessor": undefined,
  "properties": Object {
    "albums": OptionalValue$$1 {
      "defaultValue": Array [],
      "isType": true,
      "name": "Album[]",
      "type": ArrayType$$1 {
        "flags": 32,
        "isType": true,
        "name": "Album[]",
        "shouldAttachNode": true,
        "subType": ModelType$$1 {
          "flags": 128,
          "identifierAttribute": undefined,
          "initializers": Array [],
          "isType": true,
          "name": "Album",
          "properties": Object {
            "id": CoreType$$1 {
              "checker": [Function],
              "flags": 2,
              "initializer": [Function],
              "isType": true,
              "name": "number",
              "shouldAttachNode": false,
            },
            "photos": OptionalValue$$1 {
              "defaultValue": Array [],
              "isType": true,
              "name": "Photo[]",
              "type": ArrayType$$1 {
                "flags": 32,
                "isType": true,
                "name": "Photo[]",
                "shouldAttachNode": true,
                "subType": ModelType$$1 {
                  "flags": 128,
                  "identifierAttribute": undefined,
                  "initializers": Array [],
                  "isType": true,
                  "name": "Photo",
                  "properties": Object {
                    "albumId": CoreType$$1 {
                      "checker": [Function],
                      "flags": 2,
                      "initializer": [Function],
                      "isType": true,
                      "name": "number",
                      "shouldAttachNode": false,
                    },
                    "id": CoreType$$1 {
                      "checker": [Function],
                      "flags": 2,
                      "initializer": [Function],
                      "isType": true,
                      "name": "number",
                      "shouldAttachNode": false,
                    },
                    "thumbnailUrl": OptionalValue$$1 {
                      "defaultValue": "",
                      "isType": true,
                      "name": "string",
                      "type": CoreType$$1 {
                        "checker": [Function],
                        "flags": 1,
                        "initializer": [Function],
                        "isType": true,
                        "name": "string",
                        "shouldAttachNode": false,
                      },
                    },
                    "title": OptionalValue$$1 {
                      "defaultValue": "",
                      "isType": true,
                      "name": "string",
                      "type": CoreType$$1 {
                        "checker": [Function],
                        "flags": 1,
                        "initializer": [Function],
                        "isType": true,
                        "name": "string",
                        "shouldAttachNode": false,
                      },
                    },
                    "url": OptionalValue$$1 {
                      "defaultValue": "",
                      "isType": true,
                      "name": "string",
                      "type": CoreType$$1 {
                        "checker": [Function],
                        "flags": 1,
                        "initializer": [Function],
                        "isType": true,
                        "name": "string",
                        "shouldAttachNode": false,
                      },
                    },
                  },
                  "propertyNames": Array [
                    "albumId",
                    "id",
                    "title",
                    "url",
                    "thumbnailUrl",
                  ],
                  "shouldAttachNode": true,
                },
              },
            },
            "title": CoreType$$1 {
              "checker": [Function],
              "flags": 1,
              "initializer": [Function],
              "isType": true,
              "name": "string",
              "shouldAttachNode": false,
            },
            "userId": CoreType$$1 {
              "checker": [Function],
              "flags": 2,
              "initializer": [Function],
              "isType": true,
              "name": "number",
              "shouldAttachNode": false,
            },
          },
          "propertyNames": Array [
            "id",
            "userId",
            "title",
            "photos",
          ],
          "shouldAttachNode": true,
        },
      },
    },
    "posts": OptionalValue$$1 {
      "defaultValue": Array [],
      "isType": true,
      "name": "Post[]",
      "type": ArrayType$$1 {
        "flags": 32,
        "isType": true,
        "name": "Post[]",
        "shouldAttachNode": true,
        "subType": ModelType$$1 {
          "flags": 128,
          "identifierAttribute": undefined,
          "initializers": Array [],
          "isType": true,
          "name": "Post",
          "properties": Object {
            "body": CoreType$$1 {
              "checker": [Function],
              "flags": 1,
              "initializer": [Function],
              "isType": true,
              "name": "string",
              "shouldAttachNode": false,
            },
            "comments": OptionalValue$$1 {
              "defaultValue": Array [],
              "isType": true,
              "name": "Comments[]",
              "type": ArrayType$$1 {
                "flags": 32,
                "isType": true,
                "name": "Comments[]",
                "shouldAttachNode": true,
                "subType": ModelType$$1 {
                  "flags": 128,
                  "identifierAttribute": undefined,
                  "initializers": Array [],
                  "isType": true,
                  "name": "Comments",
                  "properties": Object {
                    "body": OptionalValue$$1 {
                      "defaultValue": "",
                      "isType": true,
                      "name": "string",
                      "type": CoreType$$1 {
                        "checker": [Function],
                        "flags": 1,
                        "initializer": [Function],
                        "isType": true,
                        "name": "string",
                        "shouldAttachNode": false,
                      },
                    },
                    "email": OptionalValue$$1 {
                      "defaultValue": "",
                      "isType": true,
                      "name": "string",
                      "type": CoreType$$1 {
                        "checker": [Function],
                        "flags": 1,
                        "initializer": [Function],
                        "isType": true,
                        "name": "string",
                        "shouldAttachNode": false,
                      },
                    },
                    "id": CoreType$$1 {
                      "checker": [Function],
                      "flags": 2,
                      "initializer": [Function],
                      "isType": true,
                      "name": "number",
                      "shouldAttachNode": false,
                    },
                    "name": OptionalValue$$1 {
                      "defaultValue": "",
                      "isType": true,
                      "name": "string",
                      "type": CoreType$$1 {
                        "checker": [Function],
                        "flags": 1,
                        "initializer": [Function],
                        "isType": true,
                        "name": "string",
                        "shouldAttachNode": false,
                      },
                    },
                    "postId": CoreType$$1 {
                      "checker": [Function],
                      "flags": 2,
                      "initializer": [Function],
                      "isType": true,
                      "name": "number",
                      "shouldAttachNode": false,
                    },
                  },
                  "propertyNames": Array [
                    "postId",
                    "id",
                    "name",
                    "email",
                    "body",
                  ],
                  "shouldAttachNode": true,
                },
              },
            },
            "id": CoreType$$1 {
              "checker": [Function],
              "flags": 2,
              "initializer": [Function],
              "isType": true,
              "name": "number",
              "shouldAttachNode": false,
            },
            "title": CoreType$$1 {
              "checker": [Function],
              "flags": 1,
              "initializer": [Function],
              "isType": true,
              "name": "string",
              "shouldAttachNode": false,
            },
            "userId": CoreType$$1 {
              "checker": [Function],
              "flags": 2,
              "initializer": [Function],
              "isType": true,
              "name": "number",
              "shouldAttachNode": false,
            },
          },
          "propertyNames": Array [
            "id",
            "userId",
            "title",
            "body",
            "comments",
          ],
          "shouldAttachNode": true,
        },
      },
    },
    "toggle": OptionalValue$$1 {
      "defaultValue": false,
      "isType": true,
      "name": "boolean",
      "type": CoreType$$1 {
        "checker": [Function],
        "flags": 4,
        "initializer": [Function],
        "isType": true,
        "name": "boolean",
        "shouldAttachNode": false,
      },
    },
  },
  "propertyNames": Array [
    "albums",
    "posts",
    "toggle",
  ],
  "shouldAttachNode": true,
}
`;
